ratpack-pac4j/src/test/groovy/ratpack/pac4j/AuthenticationSpec.groovy

```diff
@@ -94,4 +127,45 @@ class AuthenticationSpec extends RatpackGroovyDslSpec {
     then:
     resp2.statusCode == 200
   }
+
+  def "check authorization with direct auth"() {
+    given:
+    serverConfig {
+      development true
+    }
+    bindings {
+      module SessionModule
+    }
+    handlers {
+      all RatpackPac4j.authenticator(new DirectBasicAuthClient(new SimpleTestUsernamePasswordAuthenticator()))
+      prefix("allowed") {
+        all RatpackPac4j.requireAuth(DirectBasicAuthClient, { ctx, p -> true } as Authorizer)
+        get {
+          render "Hello " + get(UserProfile).getId()
+        }
+      }
+      prefix("forbidden") {
+        all RatpackPac4j.requireAuth(DirectBasicAuthClient, { ctx, p -> false } as Authorizer)
+        get {
+          render "You're not allowed here " + get(UserProfile).getId()
+        }
+      }
+    }
+    httpClient.requestSpec { r ->
+      r.redirects 0
+    }
+
+    expect:
+    get('allowed').statusCode == 401
+    requestSpec { r ->
+      r.basicAuth('user', 'user')
+    }.getText('allowed') == 'Hello user'
+
+    and:
+    resetRequest()
```

Comment:
Pretty easy to miss that this is necessary when doing multiple requests in one feature method. If you don't the following request inherits the basic auth header from the earlier one.